plugin/trino-exchange-filesystem/src/main/java/io/trino/plugin/exchange/filesystem/FileSystemExchange.java

```diff
@@ -186,10 +194,11 @@ private List<ExchangeSourceHandle> createExchangeSourceHandles()
         synchronized (this) {
             finishedTaskPartitions = ImmutableList.copyOf(finishedSinks);
```

Comment:
Depends. If we only have 1 partitions, it will take 40ms - 50ms; if we have 100 partitions, it will take around 200ms - 300ms, if we have 1600 partitions, it will take around 3s - 4s. Given that we need to make API calls to the cloud anyways, maybe we should still keep it?